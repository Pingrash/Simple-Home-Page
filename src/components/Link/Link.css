.link-button {
  display: flex;
  flex-direction: column;
  width: 180px;
  height: 180px;
  align-items: center;
  justify-content: center;
  background-color: #fff;
  text-decoration: none;
  color: black;
  margin: 1rem;
  box-shadow: 2px 4px lightgrey;
}

.link-button:hover {
  background-color: #f6f6f6;
}

.link-title {
  flex: 1;
  text-align: center;
  text-transform: capitalize;
  line-height: 60px;
  font-size: 2rem;
  padding-top: 1rem;
}

.link-favicon {
  flex: 1;
  width: 1rem;
  padding: 3rem;
}
