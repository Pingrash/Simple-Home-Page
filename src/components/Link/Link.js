import React from 'react';
import PropTypes from 'prop-types';
import './Link.css';

const Link = ({ link, favicon, name }) => {
  return (
    <a
      href={link}
      target='_blank'
      rel='noopener noreferrer'
      data-test='link-button'
      className='link-button'
    >
      <span className='link-title'>{name}</span>
      <img
        src={favicon}
        alt='favicon'
        data-test='favicon'
        className='link-favicon'
      />
    </a>
  );
};

Link.propTypes = {
  link: PropTypes.string.isRequired,
  favicon: PropTypes.string,
  name: PropTypes.string.isRequired
};

export default Link;
