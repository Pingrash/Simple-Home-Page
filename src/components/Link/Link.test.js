import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Link from './Link';
import { findByDataTest, checkProps } from '../../../Utilities';

const props = { link: 'link', favicon: 'favicon' };

const setup = () => {
  const component = shallow(<Link {...props} />);
  return component;
};

describe('Snapshot', () => {
  const tree = renderer.create(<Link {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Link component renders correctly', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Renders without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Renders a link button', () => {
    const link = findByDataTest(wrapper, 'link-button');
    expect(link.length).toBe(1);
  });

  it('Renders a favicon', () => {
    const favicon = findByDataTest(wrapper, 'favicon');
    expect(favicon.length).toBe(1);
  });
});

describe('Link component has correct props', () => {
  it('Should not have any prop errors', () => {
    const expectedProps = {
      link: 'link1',
      favicon: 'favicon1'
    };
    const propsError = checkProps(Link, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
