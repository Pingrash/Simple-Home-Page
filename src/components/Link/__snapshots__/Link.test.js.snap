// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<a
  className="link-button"
  data-test="link-button"
  href="link"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    className="link-title"
  >
    name
  </span>
  <img
    alt="favicon"
    className="link-favicon"
    data-test="favicon"
    src="favicon"
  />
</a>
`;
